include/nbl/video/utilities/CArithmeticOps.h

```diff
@@ -0,0 +1,286 @@
+        {
+            auto system = m_device->getPhysicalDevice()->getSystem();
+            core::smart_refctd_ptr<const system::IFile> hlsl;
+            {
+                auto loadBuiltinData = [&](const std::string _path) -> core::smart_refctd_ptr<const nbl::system::IFile>
+                {
+                    nbl::system::ISystem::future_t<core::smart_refctd_ptr<nbl::system::IFile>> future;
+                    system->createFile(future, system::path(_path), core::bitflag(nbl::system::IFileBase::ECF_READ) | nbl::system::IFileBase::ECF_MAPPABLE);
+                    if (future.wait())
+                        return future.copy();
+                    return nullptr;
+                };
+
+                // hlsl = loadBuiltinData("nbl/builtin/hlsl/scan/indirect_reduce.hlsl"); ??
+                hlsl = loadBuiltinData(shaderFile);
+            }
+            
+            auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(hlsl->getSize());
+            memcpy(buffer->getPointer(), hlsl->getMappedPointer(), hlsl->getSize());
+            auto cpushader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(buffer), asset::IShader::ESS_COMPUTE, asset::IShader::E_CONTENT_TYPE::ECT_HLSL, "????");
+
+            // REVIEW: Probably combine the commons parts of CReduce and CScanner
+            const char* storageType = nullptr;
+            switch (dataType)
+            {
+                case EDT_UINT:
+                    storageType = "uint32_t";
+                    break;
+                case EDT_INT:
+                    storageType = "int32_t";
+                    break;
+                case EDT_FLOAT:
+                    storageType = "float32_t";
+                    break;
+                default:
+                    assert(false);
+                    break;
+            }
+
+            const char* binop = nullptr;
+            switch (op)
+            {
+                case EO_AND:
+                    binop = "bit_and";
+                    break;
+                case EO_OR:
+                    binop = "bit_or";
+                    break;
+                case EO_XOR:
+                    binop = "bit_xor";
+                    break;
+                case EO_ADD:
+                    binop = "plus";
+                    break;
+                case EO_MUL:
+                    binop = "multiplies";
+                    break;
+                case EO_MAX:
+                    binop = "maximum";
+                    break;
+                case EO_MIN:
+                    binop = "minimum";
+                    break;
+                default:
+                    assert(false);
+                    break;
+            }
+            
+			return asset::CHLSLCompiler::createOverridenCopy(cpushader.get(), "#define WORKGROUP_SIZE %d\ntypedef %s Storage_t;\n#define BINOP nbl::hlsl::%s\n#define SCRATCH_SZ %d\n", m_workgroupSize, storageType, binop, scratchSz);
```

Comment:
it would be better to build the shader source instead of overriding some random shader